File: CS/PrismOnDXDocking/Bootstrapper.cs
using Microsoft.VisualBasic;
using System.Windows;
using DevExpress.Xpf.Docking;
using Microsoft.Practices.Prism.MefExtensions;
using Microsoft.Practices.Prism.Regions;
using PrismOnDXDocking.Infrastructure;
using PrismOnDXDocking.Infrastructure.Adapters;
using PrismOnDXDocking.Infrastructure.Behaviors;

namespace PrismOnDXDocking {
	public class Bootstrapper : MefBootstrapper {
		protected override void ConfigureAggregateCatalog() {
			AggregateCatalog.Catalogs.Add(new System.ComponentModel.Composition.Hosting.AssemblyCatalog(typeof(Bootstrapper).Assembly));
			AggregateCatalog.Catalogs.Add(new System.ComponentModel.Composition.Hosting.AssemblyCatalog(typeof(RegionNames).Assembly));
			AggregateCatalog.Catalogs.Add(new System.ComponentModel.Composition.Hosting.AssemblyCatalog(typeof(ExampleModule.ExampleModule).Assembly));
		}
		protected override DependencyObject CreateShell() {
			return Container.GetExportedValue<Shell>();
		}
		protected override void InitializeShell() {
			base.InitializeShell();
			Application.Current.MainWindow = (Shell)Shell;
			Application.Current.MainWindow.Show();
		}
		protected override RegionAdapterMappings ConfigureRegionAdapterMappings() {
			RegionAdapterMappings mappings = base.ConfigureRegionAdapterMappings();
            mappings.RegisterMapping(typeof(LayoutPanel), Container.GetExportedValue<LayoutPanelAdapter>());
            mappings.RegisterMapping(typeof(LayoutGroup), Container.GetExportedValue<LayoutGroupAdapter>());
            mappings.RegisterMapping(typeof(DocumentGroup), Container.GetExportedValue<DocumentGroupAdapter>());
            //mappings.RegisterMapping(typeof(TabbedGroup), Container.GetExportedValue<TabbedGroupAdapter>());
            return mappings;
		}
	}
}

File: VB/PrismOnDXDocking/Bootstrapper.vb
Imports System.Windows
Imports DevExpress.Xpf.Docking
Imports PrismOnDXDocking.Infrastructure
Imports PrismOnDXDocking.Infrastructure.Adapters
Imports Prism.Mef
Imports Prism.Regions

Namespace PrismOnDXDocking

    Public Class Bootstrapper
        Inherits MefBootstrapper

        Protected Overrides Sub ConfigureAggregateCatalog()
            AggregateCatalog.Catalogs.Add(New System.ComponentModel.Composition.Hosting.AssemblyCatalog(GetType(Bootstrapper).Assembly))
            AggregateCatalog.Catalogs.Add(New System.ComponentModel.Composition.Hosting.AssemblyCatalog(GetType(RegionNames).Assembly))
            AggregateCatalog.Catalogs.Add(New System.ComponentModel.Composition.Hosting.AssemblyCatalog(GetType(ExampleModule.ExampleModule).Assembly))
        End Sub

        Protected Overrides Function CreateShell() As DependencyObject
            Return Container.GetExportedValue(Of Shell)()
        End Function

        Protected Overrides Sub InitializeShell()
            MyBase.InitializeShell()
            Application.Current.MainWindow = CType(Shell, Shell)
            Call Application.Current.MainWindow.Show()
        End Sub

        Protected Overrides Function ConfigureRegionAdapterMappings() As RegionAdapterMappings
            Dim mappings As RegionAdapterMappings = MyBase.ConfigureRegionAdapterMappings()
            mappings.RegisterMapping(GetType(LayoutPanel), Container.GetExportedValue(Of LayoutPanelAdapter)())
            mappings.RegisterMapping(GetType(LayoutGroup), Container.GetExportedValue(Of LayoutGroupAdapter)())
            mappings.RegisterMapping(GetType(DocumentGroup), Container.GetExportedValue(Of DocumentGroupAdapter)())
            Return mappings
        End Function
    End Class
End Namespace
